core/cache-handler.mjs

```diff
@@ -1,11 +1,88 @@
-/**
- * Solution taken from here: https://github.com/vercel/next.js/discussions/48324#discussioncomment-10542097
- *
- * We are reexporting `next.js`'s default cache handler to get around
- * the fixed 2mb limit for cached fetches.
- * We run into this limit somewhat quickly if we fetch more than 100 pubs.
- */
+// Based on https://github.com/fortedigital/nextjs-cache-handler#full-example
 
-import FileSystemCache from "next/dist/server/lib/incremental-cache/file-system-cache.js";
+import { PHASE_PRODUCTION_BUILD } from "next/constants.js";
+import createBufferStringHandler from "@fortedigital/nextjs-cache-handler/buffer-string-decorator";
+import { Next15CacheHandler } from "@fortedigital/nextjs-cache-handler/next-15-cache-handler";
+import createRedisHandler from "@fortedigital/nextjs-cache-handler/redis-strings";
+import { CacheHandler } from "@neshca/cache-handler";
+import createLruHandler from "@neshca/cache-handler/local-lru";
+import { createClient } from "redis";
 
-export default FileSystemCache;
+// Usual onCreation from @neshca/cache-handler
+CacheHandler.onCreation(() => {
+	// Important - It's recommended to use global scope to ensure only one Redis connection is made
+	// This ensures only one instance get created
+	if (global.cacheHandlerConfig) {
+		return global.cacheHandlerConfig;
+	}
+
+	// Important - It's recommended to use global scope to ensure only one Redis connection is made
+	// This ensures new instances are not created in a race condition
+	if (global.cacheHandlerConfigPromise) {
+		return global.cacheHandlerConfigPromise;
+	}
+
+	// Main promise initializing the handler
+	global.cacheHandlerConfigPromise = (async () => {
+		/** @type {import("redis").RedisClientType | null} */
+		let redisClient = null;
+		if (PHASE_PRODUCTION_BUILD !== process.env.NEXT_PHASE) {
+			try {
+				redisClient = createClient({
+					url: process.env.VALKEY_URL,
+					pingInterval: 10000,
+				});
+				redisClient.on("error", (e) => {
+					if (typeof process.env.NEXT_PRIVATE_DEBUG_CACHE !== "undefined") {
```

Comment:
maybe set this to be `if (process.env.CACHE_LOG) {` ? cuts down on one new env var

Reply:
nvm, this seems to be used internally in `@neshca/cache-handler` as well, let's leave it like this